core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1599,4 +1696,31 @@ object KafkaZkClient {
       time, metricGroup, metricType)
     new KafkaZkClient(zooKeeperClient, isSecure, time)
   }
+
+
+  private def controllerZkVersionCheck(version: Int): Option[ZkVersionCheck] = {
+    if (version < 0)
+      None
+    else
+      Some(ZkVersionCheck(ControllerEpochZNode.path, version))
+  }
+
+  private def maybeThrowControllerMoveException(response: AsyncResponse): Unit = {
+    response.zkVersionCheckResult match {
+      case Some(zkVersionCheckResult) =>
+        val zkVersionCheck = zkVersionCheckResult.zkVersionCheck
+        if (zkVersionCheck.checkPath.equals(ControllerEpochZNode.path))
```

Comment:
Is it possible for `ControllerEpochZNode.path` to be different from `zkVersionCheck.checkPath`?

Reply:
Currently no. The check here is for general purpose and safety because the zkVersionCheck can apply on any znode if needed.